my_list.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'my_enumerable'
+
+class MyList
+  include MyEnumerable
+
+  def initialize(*elements)
+    @list = elements
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
+end
```

Comment:
- [Optional] It would be great to check and exit fi there is no block given for the current process.
> Something like 
```ruby 
return to_enum(:each) unless block_given?
```
- [ ] You did great team, but notice here, you are using the already in place method `each` on the list, this prevents you from experienceing the full lesson of creating an enumerator. we need to use `yiel` and continue processing for each block inside a loop. 
> Something like ...
```ruby
i=0
while i< @list.length
      yield(@list[i])
      i+= 1
    end
```
Finally, you can return your list.

Reply:
OK @TadesseDev